osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/factories/Hls5Connector.java

```diff
@@ -70,10 +71,12 @@ public DlmsConnection connect(final DlmsDevice device, final DlmsMessageListener
             throw new TechnicalException(ComponentType.PROTOCOL_DLMS,
                     "The IP address is not found: " + device.getIpAddress());
         } catch (final IOException e) {
-            if (device.hasNewSecurityKey()) {
-                // Queue key recovery process.
-                this.recoverKeyProcessInitiator.initiate(device.getDeviceIdentification(), device.getIpAddress());
+
+            //TODO: use isNew instead of isActivated (because secrets can be expired also)
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1135)